src/main/java/Deadline.java

```diff
@@ -0,0 +1,55 @@
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Representation of a Deadline task. Inherits from Task.
+ */
+public class Deadline extends Task {
+    String deadLine;
+    LocalDate localDate;
+    LocalTime localTime;
+
+    /**
+     * Class constructor. Creates a not done Deadline object with name set to the specified taskName
+     * and deadline set to the specified deadLine. Converts and stores the deadline as LocalDate and LocalTime.
+     * @param taskName The specified name of the new Deadline object.
+     * @param deadLine The specified deadline of the new Deadline object.
+     */
+    public Deadline(String taskName, String deadLine) {
+        super(taskName);
+        this.deadLine = deadLine;
+
+        String[] deadLineSplit = deadLine.split(" ");
+
+        localDate = LocalDate.parse(deadLineSplit[0]);
+
+        if (deadLineSplit.length > 1) {
+            int time = Integer.parseInt(deadLineSplit[1]);
+            localTime = LocalTime.of(time/100, time % 100);
+        } else {
+            localTime = LocalTime.MIDNIGHT;
+            this.deadLine += " 0000";
+        }
```

Comment:
Huge assumption, might tell the UI to print like "no deadline given, midnight is set to be the deadline time"